// hedron/lib/components/Base.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _noTags = require("styled-components/no-tags");

var _noTags2 = _interopRequireDefault(_noTags);

var _helpers = require("../helpers");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Base = function Base(element) {
  return (0, _noTags2.default)(element)(_helpers.generateStyles);
};

Base.propTypes = {
  // Positioning
  padding: _propTypes2.default.string,
  margin: _propTypes2.default.string,
  width: _propTypes2.default.string,
  height: _propTypes2.default.string,
  // Display
  visibility: _propTypes2.default.string,
  display: _propTypes2.default.string,
  opacity: _propTypes2.default.number,
  // Coloring
  color: _propTypes2.default.string,
  background: _propTypes2.default.string,
  border: _propTypes2.default.string,
  // Text
  fontSize: _propTypes2.default.string,
  fontWeight: _propTypes2.default.oneOf([_propTypes2.default.string, _propTypes2.default.number]),
  fontStyle: _propTypes2.default.string,
  fontFamily: _propTypes2.default.string,
  lineHeight: _propTypes2.default.number,
  textTransform: _propTypes2.default.string,
  // Convenience Properties
  hidden: _propTypes2.default.bool
};

exports.default = Base;
