// hedron/lib/components/Bounds.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Bounds = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _noTags = require("styled-components/no-tags");

var _noTags2 = _interopRequireDefault(_noTags);

var _helpers = require("../helpers");

var _Provider = require("./Provider");

var _Base = require("./Base");

var _Base2 = _interopRequireDefault(_Base);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * Bounds functions as a container to wrap `Box` components.
 * The only components that should be a direct children of `Bounds`
 * are `Box` or `Bounds`.
 */
var Bounds = exports.Bounds = (0, _noTags2.default)((0, _Base2.default)("div")).withConfig({
  displayName: "Bounds",
  componentId: "lnedi2-0"
})(["box-sizing:border-box;display:flex;", ";", ";", ";"], function (props) {
  return props.debug && (0, _helpers.drawDebug)();
}, function (props) {
  return (0, _helpers.flex)(props);
}, function (props) {
  return (0, _helpers.breakpoints)(props);
});

Bounds.propTypes = _extends({
  /**
   * Enable if you want the geometry of the grid exposed
   */
  debug: _propTypes2.default.bool,

  /**
   * Interface for setting the `flex` css property
   */
  flex: _propTypes2.default.string,

  /**
   * The primary axis the children should be in line with
   */
  direction: _propTypes2.default.oneOf(["horizontal", "vertical"]),

  /**
   * Whether children should wrap when there's no more room
   * on the primary axis.
   */
  wrap: _propTypes2.default.bool,

  /**
   * Alignment of children along the vertical axis
   */
  valign: _propTypes2.default.oneOf(["top", "center", "bottom", null]),

  /**
   * Alignment of children along the horizontal axis
   */
  halign: _propTypes2.default.oneOf(["left", "center", "right", null])
}, _Base2.default.propTypes);

Bounds.defaultProps = _extends({
  debug: false,
  direction: "horizontal",
  wrap: false
}, _Base2.default.defaultProps);

exports.default = (0, _Provider.withBreakpoints)(Bounds);
