// hedron/lib/components/Box.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Box = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _propTypes = require("prop-types");

var _propTypes2 = _interopRequireDefault(_propTypes);

var _noTags = require("styled-components/no-tags");

var _noTags2 = _interopRequireDefault(_noTags);

var _helpers = require("../helpers");

var _Provider = require("./Provider");

var _Base = require("./Base");

var _Base2 = _interopRequireDefault(_Base);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Box = exports.Box = (0, _noTags2.default)((0, _Base2.default)("div")).withConfig({
  displayName: "Box",
  componentId: "sc-195aq0d-0"
})(["", " box-sizing:border-box;", ";", ";", ";"], function (props) {
  return props.fluid && "padding: 0;";
}, function (props) {
  return props.debug && (0, _helpers.drawDebug)();
}, function (props) {
  return (0, _helpers.flex)(props);
}, function (props) {
  return (0, _helpers.breakpoints)(props);
});

Box.propTypes = _extends({
  // Debug Properties
  debug: _propTypes2.default.bool,
  // Flex-specific Properties
  flex: _propTypes2.default.bool,
  fill: _propTypes2.default.bool,
  fluid: _propTypes2.default.bool,
  shiftRight: _propTypes2.default.bool,
  shiftLeft: _propTypes2.default.bool,
  shiftUp: _propTypes2.default.bool,
  shiftDown: _propTypes2.default.bool
}, _Base2.default.propTypes);

Box.defaultProps = _extends({
  debug: false,
  fill: false
}, _Base2.default.defaultProps);

exports.default = (0, _Provider.withBreakpoints)(Box);
